modules/contact/src/constraints/ComputeFrictionalForceCartesianLMMechanicalContact.C

```diff
@@ -327,24 +331,27 @@ ComputeFrictionalForceCartesianLMMechanicalContact::enforceConstraintOnDof(
   else if (std::abs(nz) > threshold_for_Jacobian)
     component_normal = 2;
 
-  libmesh_ignore(component_normal);
-
-  _assembly.processResidualAndJacobian(
-      normal_dof_residual,
-      component_normal == 0 ? dof_index_x : (component_normal == 1 ? dof_index_y : dof_index_z),
-      _vector_tags,
-      _matrix_tags);
-
-  _assembly.processResidualAndJacobian(
-      tangential_dof_residual,
-      (component_normal == 0 || component_normal == 2) ? dof_index_y : dof_index_x,
-      _vector_tags,
-      _matrix_tags);
+  addResidualsAndJacobian(
```

Comment:
Forming the dof indices and scaling factors as an array would clean up all of the ternary logic.

Just a suggestion, I suspect that you're very much over this PR :-)

Reply:
Can you show what you're thinking? I don't get it yet.

Reply:
Not relevant to this PR; let's get this in instead.